cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -581,13 +586,57 @@ public UserTaskInfo setState(TaskState nextState) {
       return this;
     }
 
+    public CruiseControlParameters parameters() {
+      return _parameters;
+    }
+
     boolean isUserTaskDone() {
       return _futures.get(_futures.size() - 1).isDone();
     }
 
     boolean isUserTaskDoneExceptionally() {
       return _futures.get(_futures.size() - 1).isCompletedExceptionally();
     }
+
+    @Override
+    public String toString() {
+      return String.format("UserTask [%s] for endpoint [%s] with requestUrl [%s] started-at [%s]. UserTask has %d step(s).",
+                           userTaskId().toString(), endPoint().toString(), requestWithParams(),
+                           KafkaCruiseControlUtils.toDateString(startMs(), DATE_FORMAT, TIME_ZONE),
+                           futures().size());
+    }
+  }
+
+  // Persist user request and result to log. Called when the user task completes and is moved into COMPLETED status
+  private void persistCompletedUserTask(UserTaskInfo userTaskInfo) {
+    String userTaskId = userTaskInfo.userTaskId().toString();
+    List<OperationFuture> futures = userTaskInfo.futures();
+
+    RESULT_LOG.info(userTaskInfo.toString());
+    StringBuilder message = new StringBuilder();
+    int step = 1;
+    for (OperationFuture future : futures) {
+      try {
+        try {
+          AbstractCruiseControlResponse response = ((AbstractCruiseControlResponse) future.get());
+          if (userTaskInfo.parameters() != null) {
+            String cachedResponse = response.cachedResponse(userTaskInfo.parameters());
+            message.append(String.format("Result for step %d:%n", step++))
+                   .append(cachedResponse)
+                   .append("%n");
+          } else {
+            message.append("UserTask parameter is null, cannot obtain result.");
+          }
+        } catch (Exception e) {
+          // Log Exception from completeExceptionally.
+          message.append(String.format("UserTask [%s] had exception:[%s].%n", userTaskId, e.getMessage()));
+        }
+      } catch (Throwable t) {
```

Comment:
I feel the outer try block is unnecessary.

Reply:
removed